codex/stores/maintenance.nim

```diff
@@ -75,11 +75,11 @@ proc runBlockCheck(self: BlockMaintainer): Future[void] {.async.} =
     return
 
   var numberReceived = 0
-  for maybeBeFuture in iter:
-    if be =? await maybeBeFuture:
-      inc numberReceived
-      await self.processBlockExpiration(be)
-      await sleepAsync(50.millis)
+  for beFut in iter:
+    let be = await beFut
+    inc numberReceived
+    await self.processBlockExpiration(be)
```

Comment:
instead of deleting one by one, we should accomulate a batch of blocks and delete. Something for a future PR.

Reply:
Block expiration is going to be a property of a dataset if this PR goes through https://github.com/codex-storage/nim-codex/pull/678

Regardless of that deleting in a batch may be incorrect (depending on the implementation). In above PR we're still deleting blocks one by one because each block needs a careful inspection of ref-count. Delete happens only if block has ref-count = 0.